src/vm/functions/conversions.rs

```diff
@@ -113,3 +115,107 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
+            let flat = data.into_iter().flatten().collect();
+            let as_string = String::from_utf8(flat).unwrap();
+            return conversion_fn(as_string);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![
+                    TypeSignature::max_string_ascii(),
+                    TypeSignature::max_string_utf8(),
+                ],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    fn convert_to_int(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<i128>();
+        match possible_int {
+            Ok(val) => return Ok(Value::Int(val)),
+            Err(error) => return Err(CheckErrors::InvalidCharactersDetected.into()),
+        }
+    }
+    return native_string_to_int_generic(value, convert_to_int);
+}
+
+pub fn native_string_to_uint(value: Value) -> Result<Value> {
+    fn convert_to_uint(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<u128>();
+        match possible_int {
+            Ok(val) => return Ok(Value::UInt(val)),
+            Err(error) => return Err(CheckErrors::InvalidCharactersDetected.into()),
+        }
+    }
+    return native_string_to_int_generic(value, convert_to_uint);
+}
+
+pub fn native_int_to_string_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Value,
+) -> Result<Value> {
+    match value {
+        Value::Int(ref int_value) => {
+            let as_string = int_value.to_string();
+            let value = conversion_fn(as_string);
+            return Ok(value);
+        }
+        Value::UInt(ref uint_value) => {
+            let as_string = uint_value.to_string();
+            let value = conversion_fn(as_string);
+            return Ok(value);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![TypeSignature::IntType, TypeSignature::UIntType],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_int_to_ascii(value: Value) -> Result<Value> {
+    // Given a string representing an integer, convert this to Clarity ASCII value.
+    fn convert_to_ascii(raw_string: String) -> Value {
+        let data_vec = raw_string.as_bytes().to_vec();
+        return Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+            data: data_vec,
+        })));
+    }
+    return native_int_to_string_generic(value, convert_to_ascii);
+}
+
+pub fn native_int_to_utf8(value: Value) -> Result<Value> {
+    // Given a string representing an integer, convert this to Clarity UTF8 value.
+    // The conversion of an integer into a string is going to be ASCII-compliant, therefor, we just need
+    // to wrap each individual character in another vector.
+    fn convert_to_utf8(raw_string: String) -> Value {
```

Comment:
Instead of creating a bespoke consensus-critical conversion routine from a utf8 literal to a `CharType::UTF8()`, can you just use the existing `Value::string_utf8_from_string_utf8_literal()`?

Reply:
is it possible to just use `Value::string_utf8_from_bytes` here, because we know the output of "int -> string" will be within the ASCII subset?

Also, in case this call fails, I have added a new InternalError to the erros in `src/vm/analysis/errors.rs`. Do you think that is appropriate?

Reply:
Note: I removed InternalError.